includes/class-sensei.php

```diff
@@ -918,80 +919,203 @@ public function load_class( $class_name = '' ) {
+	}
+
+	/**
+	 * Clear the comment count cache.
+	 *
+	 * @param int $post_id Post ID.
+	 */
+	public function flush_comment_counts_cache( $post_id ) {
+		$post_id = (int) $post_id;
+
+		$post_transient_id = self::COMMENT_COUNT_TRANSIENT_PREFIX . $post_id;
+		delete_transient( $post_transient_id );
+
+		$all_transient_id = self::COMMENT_COUNT_TRANSIENT_PREFIX . '0';
+		delete_transient( $all_transient_id );
+	}
 
-		$total    = 0;
-		$approved = array(
-			'0'            => 'moderated',
-			'1'            => 'approved',
-			'spam'         => 'spam',
-			'trash'        => 'trash',
-			'post-trashed' => 'post-trashed',
+	/**
+	 * Get the Sensei related comment counts by comment_approved but use caching.
+	 *
+	 * @param int $post_id Post ID.
+	 *
+	 * @return array
+	 */
+	private function get_sensei_comment_counts( $post_id ) {
+		$post_id = (int) $post_id;
+
+		$transient_id = self::COMMENT_COUNT_TRANSIENT_PREFIX . $post_id;
+		$stats        = get_transient( $transient_id );
+
+		if ( ! $stats || ! is_array( $stats ) ) {
+			$stats = $this->get_sensei_comment_counts_direct( $post_id );
+
+			set_transient( $transient_id, $stats );
+		}
+
+		return $stats;
+	}
+
+	/**
+	 * Get the Sensei related comment counts by comment_approved.
+	 *
+	 * @param  int $post_id Post ID.
+	 *
+	 * @return array
+	 */
+	private function get_sensei_comment_counts_direct( $post_id ) {
+		global $wpdb;
+
+		$post_where = '';
+
+		if ( ! empty( $post_id ) ) {
+			$post_where = $wpdb->prepare( 'AND comment_post_ID=%d', $post_id );
+		}
+
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared -- `$post_where` prepared above.
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.NoCaching -- Cached in calling method.
+		$counts = $wpdb->get_results(
+			"
+					SELECT comment_approved, COUNT(*) AS num_comments
+					FROM {$wpdb->comments}
+					WHERE
+						comment_type IN ( 'sensei_course_status', 'sensei_lesson_status', 'sensei_user_answer' )
+						{$post_where}
+					GROUP BY comment_approved
+			",
+			ARRAY_A
 		);
+		// phpcs:enable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
 
-		$stats               = [];
-		$statuses_to_exclude = array( 'post-trashed', 'trash' );
+		$stats = [
+			'all'            => 0,
+			'total_comments' => 0,
+		];
 
-		foreach ( (array) $count as $row ) {
-			// Don't count post-trashed toward totals.
-			$comment_approved = $row['comment_approved'];
+		foreach ( (array) $counts as $row ) {
+			$row['num_comments'] = (int) $row['num_comments'];
 
-			if ( ! in_array( $comment_approved, $statuses_to_exclude, true ) ) {
-				$total += $row['num_comments'];
-			}
-			if ( isset( $approved[ $row['comment_approved'] ] ) ) {
-				$stats[ $approved[ $row['comment_approved'] ] ] = $row['num_comments'];
+			// Don't count post-trashed toward totals.
+			if ( ! in_array( $row['comment_approved'], array( 'post-trashed', 'trash', 'spam' ), true ) ) {
```

Comment:
I think that writing this like is a bit cleaner:
```
if ( ! in_array( $row['comment_approved'], array( 'post-trashed', 'trash' ) ) {
	$stats['total_comments'] += $row['num_comments'];
	
	if ( $row['comment_approved'] !== 'spam' ) {
		$stats['all'] += $row['num_comments'];
	}
}
```
Feel free to ignore this though as it is opinion based :)

Reply:
Yeah, that looks better. I copied this from WooCommerce to match their implementation, but that code should match.

Reply:
467558306cfc879b46706de2b4b8d083f3556adc